src/stronghold/secret.rs

```diff
@@ -142,139 +143,56 @@ impl SecretManage for StrongholdAdapter {
-            // XXX: Should we create a separate error type for this error?
-            ProcResult::BIP39Recover(ResultMessage::Error(err)) => Err(crate::Error::StrongholdProcedureError(err)),
-            // Generic Stronghold procedure failure.
-            ProcResult::Error(err) => Err(crate::Error::StrongholdProcedureError(err)),
-            // Unexpected result type, which should never happen!
-            err => {
-                warn!(
-                    "StrongholdSecretManager::bip39_recover(): unexpected result from Stronghold: {:?}",
-                    err
-                );
-                Err(crate::Error::StrongholdProcedureError(format!("{:?}", err)))
-            }
-        }
+            })?;
+
+        Ok(())
     }
 
     /// Execute [Procedure::SLIP10Derive] in Stronghold to derive a SLIP-10 private key in the Stronghold vault.
-    async fn slip10_derive(
-        &self,
-        // Stronghold 0.4.1 is still using an older version of iota-crypto, so we ask for a different one here.
-        chain: crypto05::keys::slip10::Chain,
-        input: SLIP10DeriveInput,
-        output: Location,
-        hint: RecordHint,
-    ) -> Result<()> {
-        match self
-            .stronghold
+    async fn slip10_derive(&self, chain: Chain, input: Slip10DeriveInput, output: Location) -> Result<()> {
+        self.stronghold
             .lock()
             .await
-            .runtime_exec(Procedure::SLIP10Derive {
-                chain,
-                input,
-                output,
-                hint,
-            })
-            .await
-        {
-            // SLIP-10 derivation success.
-            // We don't care about the returned value, as later we use the output in vault.
-            ProcResult::SLIP10Derive(ResultMessage::Ok(_)) => Ok(()),
-            // SLIP-10 derivation failure.
-            // XXX: Should we create a separate error type for this error?
-            ProcResult::SLIP10Derive(ResultMessage::Error(err)) => Err(crate::Error::StrongholdProcedureError(err)),
-            // Generic Stronghold procedure failure.
-            ProcResult::Error(err) => Err(crate::Error::StrongholdProcedureError(err)),
-            // Unexpected result type, which should never happen!
-            err => {
-                warn!(
-                    "StrongholdSecretManager::slip10_derive(): unexpected result from Stronghold: {:?}",
-                    err
-                );
-                Err(crate::Error::StrongholdProcedureError(format!("{:?}", err)))
-            }
-        }
+            .load_client(PRIVATE_DATA_CLIENT_PATH)?
+            .execute_procedure(procedures::Slip10Derive { chain, input, output })?;
+
+        Ok(())
     }
 
     /// Execute [Procedure::Ed25519PublicKey] in Stronghold to get an Ed25519 public key from the SLIP-10 private key
     /// located in `private_key`.
     async fn ed25519_public_key(&self, private_key: Location) -> Result<[u8; 32]> {
```

Comment:
I wonder, if we there is a way to easily export the Procedures as simple functions. But the effort here is limited, so may not be really an advantage.